packages/salesforcedx-vscode-core/src/commands/forceSourceDeploySourcePath.ts

```diff
@@ -114,28 +119,53 @@ export class LibraryDeploySourcePathExecutor extends DeployRetrieveLibraryExecut
         'Deploy (Beta)',
         'force_source_deploy_with_sourcepath_beta'
       );
-
+      channelService.showCommandWithTimestamp(`Starting ${this.executionName}`);
       if (this.sourceClient === undefined) {
         throw new Error('SourceClient is not established');
       }
-
-      this.sourceClient.tooling.deploy = this.deployWrapper(
-        this.sourceClient.tooling.deploy
-      );
-
       const projectNamespace = (await SfdxProjectConfig.getValue(
         'namespace'
       )) as string;
       const registryAccess = new RegistryAccess();
       const components = registryAccess.getComponentsFromPath(response.data);
-      const deployPromise = this.sourceClient.tooling.deploy({
-        components,
-        namespace: projectNamespace
-      });
+      let deployPromise: Promise<SourceDeployResult>;
+      if (projectNamespace) {
+        deployPromise = this.sourceClient.tooling.deploy(components, {
+          namespace: projectNamespace
+        }) as Promise<SourceDeployResult>;
+      } else {
+        deployPromise = this.sourceClient.metadata.deploy(
+          components
+        ) as Promise<SourceDeployResult>;
+      }
       const metadataCount = JSON.stringify(createComponentCount(components));
-      await deployPromise;
-
+      const result = (await vscode.window.withProgress(
+        {
+          title: this.executionName,
+          location: vscode.ProgressLocation.Notification
+        },
+        () => deployPromise
+      )) as SourceDeployResult;
+      const parser = new LibraryDeployResultParser(result);
+      const outputResult = parser.resultParser(result);
       this.logMetric({ metadataCount });
+      channelService.appendLine(outputResult);
+      channelService.showCommandWithTimestamp(`Finished ${this.executionName}`);
+      if (
+        result.status === DeployStatus.Succeeded ||
+        // @ts-ignore
+        result.status === ToolingDeployStatus.Completed
+      ) {
+        DeployRetrieveLibraryExecutor.errorCollection.clear();
+        await notificationService.showSuccessfulExecution(this.executionName);
```

Comment:
```suggestion
        notificationService.showSuccessfulExecution(this.executionName);
```

Can remove since further execution isn't dependent on this exiting

Reply:
This causes the warning "Promises must be handled appropriately (no-floating-promises)." Should I ignore this warning?

Reply:
you can append a catch block to get rid of the warning: `notificationService.showSuccessfulExecution(this.executionName).catch(error => {})`;